src/neo/Trie/MPT/MPTNode/FullNode.cs

```diff
@@ -0,0 +1,58 @@
+using System.IO;
+using Neo.Cryptography;
+using Neo.IO;
+
+namespace Neo.Trie.MPT
+{
+    public class FullNode : MPTNode
+    {
+        public MPTNode[] Children = new MPTNode[17];
+
+        public FullNode()
+        {
+            nType = NodeType.FullNode;
+            for (int i = 0; i < Children.Length; i++)
+            {
+                Children[i] = HashNode.EmptyNode();
+            }
+        }
+
+        protected override byte[] calHash()
+        {
+            var bytes = new byte[0];
+            for (int i = 0; i < Children.Length; i++)
+            {
+                bytes = bytes.Concat(Children[i].GetHash());
+            }
+            return bytes.Sha256();
+        }
+
+        public FullNode Clone()
+        {
+            var cloned = new FullNode();
+            for (int i = 0; i < Children.Length; i++)
+            {
+                cloned.Children[i] = Children[i];
+            }
+            return cloned;
+        }
+
+        public override void Serialize(BinaryWriter writer)
+        {
+            base.Serialize(writer);
+            for (int i = 0; i < Children.Length; i++)
+            {
+                writer.WriteVarBytes(Children[i].GetHash());
```

Comment:
writer.WriteVarBytes(Children[i]) ?

Reply:
Children's Hash not Children.
If we need to know the specific child, we will use the hash to resolve.